libsql-sqlite3/src/vectorfloat32.c

```diff
@@ -198,6 +198,23 @@ float vectorF32DistanceCos(const Vector *v1, const Vector *v2){
   return 1.0 - (dot / sqrt(norm1 * norm2));
 }
 
+float vectorF32DistanceL2(const Vector *v1, const Vector *v2){
+  float sum = 0;
+  float *e1 = v1->data;
+  float *e2 = v2->data;
+  int i;
+
+  assert( v1->dims == v2->dims );
+  assert( v1->type == VECTOR_TYPE_FLOAT32 );
+  assert( v2->type == VECTOR_TYPE_FLOAT32 );
+
+  for(i = 0; i < v1->dims; i++){
+    float d = e1[i]-e2[i];
```

Comment:
Would it lead to a better precision if we use double here and make sum a double as well and then cast the value to float only before returning?

Reply:
Yes, maybe. But we need to change this everywhere in the file, I guess.

I will leave for now `float` version and maybe we can change it later